pkg/storage/streams/store.go

```diff
@@ -270,11 +362,50 @@ func (lr *lazySegmentRanger) Size() int64 {
 // Range implements Ranger.Range to be lazily connected
 func (lr *lazySegmentRanger) Range(ctx context.Context, offset, length int64) (io.ReadCloser, error) {
 	if lr.ranger == nil {
-		rr, _, err := lr.segments.Get(ctx, lr.path)
+		rr, m, err := lr.segments.Get(ctx, lr.path)
+		if err != nil {
+			return nil, err
+		}
+		encryptedEncKey := m.Data
+		e, err := eestream.Decrypt(encryptedEncKey, lr.derivedKey, lr.startingNonce, lr.encType)
+		if err != nil {
+			return nil, err
+		}
+		var encKey [32]byte
+		copy(encKey[:], e)
+		decrypter, err := eestream.NewDecrypter(&encKey, lr.startingNonce, lr.encBlockSize, lr.encType)
+		if err != nil {
+			return nil, err
+		}
+
+		var rd ranger.Ranger
+		if rr.Size()%int64(decrypter.InBlockSize()) != 0 {
+			reader, err := rr.Range(ctx, 0, rr.Size())
+			if err != nil {
+				return nil, err
+			}
+			cipherData, err := ioutil.ReadAll(reader)
+			if err != nil {
+				return nil, err
+			}
+			data, err := eestream.Decrypt(cipherData, &encKey, lr.startingNonce, lr.encType)
+			if err != nil {
+				return nil, err
+			}
+			rd = ranger.ByteRanger(data)
+		} else {
+			rd, err = eestream.Transform(rr, decrypter)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		paddedSize := rd.Size()
+		rc, err := eestream.Unpad(rd, int(paddedSize-lr.Size())) // int64 -> int; is this a problem?
```

Comment:
We should unpad the data only if using the eestream block transformer, i.e. in the else block above.